MekHQ/src/mekhq/gui/dialog/InsufficientAstechTimeNagDialog.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2021 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.gui.dialog;
+
+import mekhq.MekHQ;
+import mekhq.MekHqConstants;
+import mekhq.campaign.Campaign;
+import mekhq.campaign.personnel.Person;
+import mekhq.gui.baseComponents.AbstractMHQNagDialog;
+
+import javax.swing.*;
+
+public class InsufficientAstechTimeNagDialog extends AbstractMHQNagDialog {
+    //region Constructors
+    public InsufficientAstechTimeNagDialog(final JFrame frame, final Campaign campaign) {
+        super(frame, "InsufficientAstechTimeNagDialog", "InsufficientAstechTimeNagDialog.title",
+                "", campaign, MekHqConstants.NAG_INSUFFICIENT_ASTECH_TIME);
+    }
+    //endregion Constructors
+
+    @Override
+    protected boolean checkNag(Campaign campaign) {
+        if (MekHQ.getMekHQOptions().getNagDialogIgnore(getKey())
+                || !campaign.getCampaignOptions().checkMaintenance()) {
+            return false;
+        }
+
+        final int need = campaign.getHangar().getUnitsStream()
```

Comment:
Let's put a comment explaining this syntax, as it's a little lengthy and takes some time to decode.